app/assets/javascripts/discourse/controllers/preferences/second-factor.js.es6

```diff
@@ -1,39 +1,25 @@
 import { default as computed } from "ember-addons/ember-computed-decorators";
+import CanCheckEmails from "discourse/mixins/can-check-emails";
 import { default as DiscourseURL, userPath } from "discourse/lib/url";
 import { popupAjaxError } from "discourse/lib/ajax-error";
 import { findAll } from "discourse/models/login-method";
 import { SECOND_FACTOR_METHODS } from "discourse/models/user";
+import showModal from "discourse/lib/show-modal";
 
-export default Ember.Controller.extend({
+export default Ember.Controller.extend(CanCheckEmails, {
   loading: false,
+  dirty: false,
   resetPasswordLoading: false,
   resetPasswordProgress: "",
   password: null,
-  secondFactorImage: null,
-  secondFactorKey: null,
-  showSecondFactorKey: false,
   errorMessage: null,
   newUsername: null,
   backupEnabled: Ember.computed.alias("model.second_factor_backup_enabled"),
   secondFactorMethod: SECOND_FACTOR_METHODS.TOTP,
+  totps: [],
```

Comment:
Arrays and objects defined directly on any  `Ember.Object`  are shared across all instances of that class. Try to set them in the init function.